### PR TITLE
not repeating the strings twice

### PR DESCRIPTION
**Pull request by iopq**
I was bothered by making mistakes in syncing the list of strings, so I made a macro to convert string to enum and the other way around

### CODE REVIEW

**Comment by ujh** [comment]
Have you added yourself as an author to the files you changed?

File: Cargo.lock [hunk 1]
```diff
@@ -6,8 +6,9 @@ dependencies = [
  "rand 0.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "regex 0.1.19 (registry+https://github.com/rust-lang/crates.io-index)",
  "regex_macros 0.1.12 (registry+https://github.com/rust-lang/crates.io-index)",
- "rustc-serialize 0.3.1 (registry+https://github.com/rust-lang/crates.io-index)",
- "time 0.1.19 (git+https://github.com/rust-lang/time)",
+ "rustc-serialize 0.3.3 (registry+https://github.com/rust-lang/crates.io-index)",
+ "strenum 0.0.3 (registry+https://github.com/rust-lang/crates.io-index)",
+ "time 0.1.20 (git+https://github.com/rust-lang/time)",
```
Comment (ujh) [inline Cargo.lock]: Have you tried the new time crate from crates.io? Maybe your bug is already fixed.

File: src/gtp/mod.rs [hunk 2]
```diff
@@ -30,11 +30,31 @@ use game::Game;
...
 use sgf::parser::Parser;
 use timer::Timer;
+use strenum::Strenum;
 
 pub mod driver;
 mod test;
 
-#[derive(Debug)]
+strenum! {
+    knownCommands =>
```
Comment (ujh) [inline src/gtp/mod.rs]: I'd prefer `KnownCommands` here.

File: src/gtp/mod.rs [hunk 3]
```diff
@@ -230,14 +231,6 @@ impl<'a> GTPInterpreter<'a> {
...
     fn list_commands(&self) -> String {
-        let mut result = String::new();
-
-        for c in self.known_commands.iter() {
-            result.push_str(c.as_slice());
-            result.push_str("\n");
-        }
-        result.pop();
-        result
+    	<knownCommands>::stringify()
```
Comment (ujh) [inline src/gtp/mod.rs]: :+1:

File: src/gtp/mod.rs [hunk 4]
```diff
@@ -204,14 +205,14 @@ impl<'a> GTPInterpreter<'a> {
...
-            "loadsgf" => {
+            Some(knownCommands::loadsgf) => {
                 let filename = command[1];
                 let parser = Parser::from_path(Path::new(filename));
                 let game = parser.game().unwrap();
                 self.game = game;
                 Command::LoadSgf
             },
-            _             => return Command::Error
+            None             => return Command::Error
```
Comment (ujh) [inline src/gtp/mod.rs]: Awesome! So I assume that means that the `Some` cases have to be exhaustive, and the compiler will throw an error if one is missing? That would be brilliant.

Comment (iopq) [inline src/gtp/mod.rs]: Yes, the point is that the match is exhaustive

File: Cargo.lock (hunk 1 above)
Comment (iopq) [inline Cargo.lock]: I have that in my next changeset

File: src/gtp/mod.rs (hunk 2 above)
Comment (iopq) [inline src/gtp/mod.rs]: I changed this in my other branch as well

**Comment by ujh** [comment]
OK, great! Just add yourself as an author to the files you've changed and then feel free to merge it.

**Pull request merged**